guardian/prosemirror-elements
# chore: Introduce esbuild for project build

## What does this change?

Adds esbuild to the repo, speeding up build startup and watch times considerably.

To ensure typechecking continues, we use `tork-ts-checker-webpack-plugin`, which runs in parallel with the main build and does not block module output in watch mode.

Unfortunately, esbuild doesn't support the [new jsx runtime yet](https://github.com/evanw/esbuild/issues/334), which means we need to explicitly import `React` when using jsx. (We were inconsistent before.) I've added a linting rule via the `react/recommended` plugin to ensure this is the case, and fixed a couple of other things that this new linting config spotted, too.

## How to test

Run `yarn watch` and `yarn build`. (Don't forget to run `yarn` in the project root beforehand, to install the new dependencies!) The build should work as before, and the tests should pass.

Both operations should be noticeably faster.

## How can we measure success?

Faster watch and build operations.
